bootstrap-joining-demo/kubernetes-api/src/main/scala/akka/discovery/udp/UdpBroadcastDiscovery.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.discovery.udp
+
+import java.net.InetSocketAddress
+
+import akka.actor.Actor
+import akka.actor.ActorLogging
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.pattern.ask
+import akka.discovery.Lookup
+import akka.discovery.SimpleServiceDiscovery
+import akka.discovery.SimpleServiceDiscovery.Resolved
+import akka.discovery.SimpleServiceDiscovery.ResolvedTarget
+import akka.io.Udp
+import akka.util.ByteString
+import akka.util.Timeout
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class UdpBroadcastDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
```

Comment:
The whole lifecycle of this discovery mechanism is unclear right now. The reason is probably that it mixes up the server and client side of service discovery. 

For running an active discovery (client side) we need to broadcast discovery probes (one or several ones?) and wait for a certain time to wait for responses from peers. We don't know the number of peers so we cannot easily shortcut the lookup.

We could probably leave the server side running for nodes wanting to join later without having to know explicit seed nodes.

More ideas:
 * keep track of peers and also return those on queries
 * allow to use this service discovery mechanism for any kind of service discovery

Alternatives:
 * instead of the query/response protocol (joining node is actively searching for peers), during an initial interval broadcast beacons with the own address, other peers would listen to those during the initial phase (joining node is passively waiting for node beacons). This wouldn't fit the ServiceDiscovery pattern well and also wouldn't work as well for nodes that want to join later because you probably wouldn't want to publish beacons all the time